File: Strivers/src/basicsmaths/divisors/Divisors.java
package basicsmaths.divisors;

import java.util.ArrayList;
import java.util.Collection;
import java.util.Collections;
import java.util.Scanner;

public class Divisors {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int number = sc.nextInt();
        System.out.println(sumOfDivisors(number));
    }

    static long sumOfDivisors(int N){
        long total = 0;
        for(int i = 1; i <= N; i++) {
            total += i * N / i;
        }
        return total;
    }
}

File: Strivers/src/basicsmaths/divisors/Divisors.kt
package basicsmaths.divisors

import java.util.*
import kotlin.math.sqrt

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val number = sc.nextInt()
    println(divisorsOfNumber(number).sorted())
}

private fun divisorsOfNumber(number: Int): List<Int> {
    val arrList = mutableListOf<Int>()
    var i = 1
    while (i <= sqrt(number.toDouble())) {
        if (number % i == 0) {
            arrList.add(i)
            if (i != number / i) {
                arrList.add(number / i)
            }
        }
        i++
    }
    arrList.sorted()
    return arrList
}